src/topology/category/Profinite.lean

```diff
@@ -76,3 +79,47 @@ instance : faithful Profinite_to_CompHaus := {}
 @[simp] lemma Profinite_to_CompHaus_to_Top :
   Profinite_to_CompHaus ⋙ CompHaus_to_Top = Profinite_to_Top :=
 rfl
+
+namespace Profinite
+local attribute [instance] connected_component_setoid
+
+/--
+π₀ functor from CompHaus to Profinite, quotienting a space by its connected components.
+See: https://stacks.math.columbia.edu/tag/0900
+-/
+def CompHaus_to_Profinite : CompHaus ⥤ Profinite :=
+{ obj := λ X,
+    { to_Top := { α := (π₀ X.to_Top.α) },
+      is_compact := quotient.compact_space,
+      is_t2 := pi0.t2,
+      is_totally_disconnected := pi0.totally_disconnected_space },
+  map := λ X Y f,
+    { to_fun := pi0_map f.2,
+      continuous_to_fun := pi0_map_continuous f.2 }}
+
+instance : is_right_adjoint Profinite_to_CompHaus :=
+{ left := CompHaus_to_Profinite,
+  adj :=
+  { hom_equiv := λ X Y,
+    { to_fun := λ f,
+      { to_fun := f.1 ∘ quotient.mk,
+        continuous_to_fun := continuous.comp f.2 (continuous_quotient_mk) },
+      inv_fun := λ g,
+        { to_fun := pi0_lift g.2,
+          continuous_to_fun := pi0_lift_continuous g.2 },
+      left_inv := λ f, continuous_map.ext $ λ x, quotient.induction_on x $ λ a, rfl,
+      right_inv := λ f, continuous_map.ext $ λ x, rfl },
+    unit :=
+      { app := λ X, { to_fun := quotient.mk,
+                      continuous_to_fun := continuous_quotient_mk }},
+    counit :=
+      { app := λ Y, { to_fun := pi0_lift (@continuous_map.coe_continuous _ _ _ _ (𝟙 Y.to_Top)),
+                      continuous_to_fun := { is_open_preimage := λ s hs, hs }}}}}
```

Comment:
Is it any nicer to use the constructors `adjunction.left_adjoint.of_equiv` here, so you can supply fewer fields?

Reply:
I was thinking about this whilst looking at your proof that CompHaus is reflective. Then I would have to define the functor via left_adjoint_of_equiv right? Is there nothing to gain from defining the functor directly (in term of being easier to work with?). I guess no one will use CompHaus_to_Profinite itself though as the main point is to know the other one is a right adjoint.

Reply:
You gain a tiny bit in that the `map` of the functor and the data of the adjunction are definitionally what you want, but typically you rarely need this, and in practice it usually happens that it's just `comp_id` or `id_comp` away from what's already here. The object function is the same either way round though (definitionally), and I think that's the main one you want anyway; like you say usually just the fact that we have a right adjoint is good enough. There's also the option of using the constructors `adjunction.mk_of_hom_equiv` or `adjunction.mk_of_unit_counit` which is something of an intermediate

Reply:
I rewrote it to use `adjuntion.left_adjoint.of_equiv` now. I also added the instance `reflective Profinite_to_Top`

Reply:
I had some problems with type class inference though, so the proof of `reflective Profinite_to_Top` is uglier than it needs to be...

Reply:
I think that instance isn't strictly necessary for this PR to be honest, so don't worry about it too much! More specifically, I think it should be an instance that the composition of reflective functors is reflective, so this is a gap in the API of reflective functors (thanks!)

Reply:
fixed!